code/game/objects/items/rogueweapons/rmb_intents.dm

```diff
@@ -108,3 +108,35 @@
 	name = "weak"
 	desc = "Your attacks have -1 strength and will never critically-hit. Useful for longer punishments, play-fighting, and bloodletting."
 	icon_state = "rmbweak"
+
+/datum/rmb_intent/door_unlock
+	name = "unlock"
+	desc = "Attempt to unlock a door with keys in your possession."
+	icon_state = "rmbunlock"
+
+/datum/rmb_intent/door_unlock/special_attack(mob/living/user, atom/target)
+	if(!istype(target, /obj/structure/mineral_door))
+		return
+	var/obj/structure/mineral_door/door = target
+	if(!door.keylock)
+		return
+	if(door.door_opened || door.isSwitchingStates || door.brokenstate)
+		return
+	
+	// Check user's inventory for keys
+	var/list/user_keys = user.GetAllContents(/obj/item/roguekey)
```

Comment:
I believe this goes through all of your belongings, including pouches and other containers, meaning you could unlock a door without having the keys even on a visible slot. Might be a bit too much?